### PR TITLE
Add AudioSelector and code refactoring

### PR DESCRIPTION
**Pull request by nickhsine**
@hcchien

### CODE REVIEW

File: admin/client/components/AudioGrid.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+          display: 'inline-block',
+          padding: '10px',
+          width: `${width}%`
+      }
+      return (
+          <div onClick={this._handleSelect.bind(this)} style={style}>
+              <ImageItem
+                  doShowRemove={doShowRemove}
+                  id={_.get(coverPhoto, 'id')}
+                  url={_.get(image, 'url')}
```
Comment (hcchien) [inline admin/client/components/AudioGrid.js]: do we need to check if the url or cover photo is empty?

File: admin/client/components/AudioGrid.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            selectedAudios: nextProps.selectedAudios
+        });
+    }
+    shouldComponentUpdate (nextProps, nextState) {
+        return  !shallowEqual(this.props, nextProps) ||
+            !shallowEqual(this.state, nextState);
+    }
+
+    _handleSelect (audio) {
+        console.log('click audio:', audio);
```
Comment (hcchien) [inline admin/client/components/AudioGrid.js]: keep console.log here?

File: admin/client/components/AudioSelector.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            Promise.all(promises)
+            .then((audios) => {
+                resolve(audios);
+            }, (err) => {
+                reject(err);
+            })
+        })
+    }
+
+    loadItems (querstring = '') {
```
Comment (hcchien) [inline admin/client/components/AudioSelector.js]: querstring? querystring?

**Comment by hcchien** [comment]
👍

File: admin/client/components/AudioGrid.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+                return element.id === audio.id;
+            }) ? true : false;
+            return (
+                <AudioItem
+                    audio={_.get(audio, 'url')}
+                    coverPhoto={_.get(audio, 'coverPhoto')}
+                    description={_.get(audio, 'description')}
+                    id={audio.id}
+                    isSelected={isSelected}
+                    key={audio.id}
```
Comment (hcchien) [inline admin/client/components/AudioGrid.js]: same with the id?

Comment (nickhsine) [inline admin/client/components/AudioGrid.js]: yes, ```id``` is  a ```prop```. and ```key``` is for rendering lots of components(array) in React.

File: admin/client/components/AudioSelector.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            if (imageId) {
+                this._loadImage(imageId)
+                .then((image) => {
+                    _.set(audio, [ 'fields', 'coverPhoto' ], image);
+                    resolve(parseAudioAPIResponse(audio));
+                }, (err) => {
+                    resolve(parseAudioAPIResponse(audio));
+                })
+            } else {
+                resolve(parseAudioAPIResponse(audio));
```
Comment (hcchien) [inline admin/client/components/AudioSelector.js]: maybe you can clean the code here

**Pull request merged**